lib/web_ui/lib/src/engine/text_editing/text_editing.dart

```diff
@@ -1438,7 +1438,7 @@ abstract class DefaultTextEditingStrategy with CompositionAwareMixin implements
       if (event.keyCode == _kReturnKeyCode) {
         onAction!(inputConfiguration.inputAction);
         // Prevent the browser from inserting a new line when it's not a multiline input.
-        if (inputConfiguration.inputType is! MultilineInputType) {
+        if (inputConfiguration.inputType is! MultilineInputType && inputConfiguration.inputType is! NoTextInputType) {
```

Comment:
If we end up doing this, we will need a test to make sure it doesn't break in the future. You can take a look here for an example: https://github.com/flutter/engine/blob/4a14ee8072a3d3b12b830ca996fd656a6bca774b/lib/web_ui/test/engine/text_editing_test.dart#L421

Reply:
Added test for `EngineInputType.multilineNone`. Is there a need for more tests? Thank you.